### ISSUE TITLE
MacFuse 4.5.0 does not work in Mac OS X Sonoma

### ISSUE THREAD

**Issue by alexbet** [osxfuse/osxfuse#977]
We heavily depend on MacFuse to communicate with our files in the cloud, and now this. Some of our computers were updated to Mac OS Sonoma, and MacFuse doesn't work. VERY frustrating now that we can't access our files at all.  

<img width="255" alt="Screenshot 2023-10-08 at 2 36 10 PM" src="https://github.com/osxfuse/osxfuse/assets/3966760/a25887a9-f1c7-461c-95ab-07da0ea88472">

Is there any chance that you can release a new version that works with Mac OS Sonoma? Thanks!

**Comment by bfleischer** [comment]
Version 4.5.0 works fine on macOS Sonoma. The dialog shown in the screenshot is from an older version of macFUSE (< 4.5.0). What app/file system are you trying to use on Sonoma?

**Comment by Zzuk1946** [comment]
The same thing happens to me too with Mac Studio M1 Cloud. Can anyone solve the problem?

**Comment by bfleischer** [comment]
@Zzuk1946 What file system/app are you using?

Some apps bundle custom versions of macFUSE. You might be using one of those apps. Version 4.5.0 is compatible with macOS Sonoma. The app you are using might be bundling an older version of macFUSE.

**Comment by Zzuk1946** [comment]
I've tried older versions, but Sonoma rejects them.

**Comment by bfleischer** [comment]
macFUSE 4.5.0 (or later) is required for Sonoma. Older versions of macFUSE will not work on Sonoma.

What file system/app are you using when the "Unsupported macOS Version" is displayed?

**Comment by Zzuk1946** [comment]
The prompt appears: "Open System Preferences". I open it, it tells me it's version 4.5.0. But nothing works. It doesn't seem to exist.

**Comment by bfleischer** [comment]
The error message is displayed as the result of an app trying to mount a virtual volume. Without knowing which app triggered the error message there is nothing I can do. What exactly did you do before the error message was displayed. Maybe you tried to access files in a cloud storage service or an encrypted container?

**Comment by Zzuk1946** [comment]
No. I only use it for pCloud and with Ventura it was very good. Now when I start the computer it writes to me that the system is blocked. I do everything but it doesn't unlock. I also reinstalled it several times.
Anyway, thanks for trying to help me.

**Comment by bfleischer** [comment]
pCloud might be the culprit here. Did you make sure to update pCloud to the latest version? pCloud bundles a custom version of macFUSE. Unless you updated pCloud to the latest version, you might still be using a version pCloud that does not support Sonoma.

**Comment by Zzuk1946** [comment]
I did this too.... Thank you very much for your help.

**Comment by alexbet** [comment]
I can confirm that our version was lower than 4.5.0. We installed 4.5.0, and it works fine now on Sonoma.

**Comment by ghost** [comment]
Same here. My version of macFUSE was lower than 4.5.0, but after I upgraded it, everything started working fine on macOS Sonoma.

To upgrade macFUSE to the latest version:

1. Open System Preferences.
2. Click the FUSE icon.
3. Click Update FUSE.
4. Follow the instructions to complete the upgrade.

![image](https://github.com/osxfuse/osxfuse/assets/29850594/5830beb5-2d12-4f68-b0b1-0cefe72b80c2)

**Comment by HofiOne** [comment]
I've upgraded to macFuse 4.5.0 after upgrading from macOS Ventura to Sonoma (the previous version of macFuse was already installed on Ventura)

On Ventura, macFuse worked seamlessly, but after the upgrade, it stopped working
The issue is that anytime I mount a drive it reports the same error 
`remote host has disconnected`
Tried with multiple hosts, uninstall, reinstall, did not help unfortunately :(

Any hint about this issue?

Thank you in advance!

![Screenshot 2023-10-13 at 14 40 21](https://github.com/osxfuse/osxfuse/assets/12213492/89d6eac8-a7da-4072-9e8f-291579bfa92b)

**Comment by Nathan187** [comment]
there are still issues with veracrypt and macfuse.

**Comment by mnrv-ae** [comment]
Just today I've made the mistake of upgrading to Sonoma 14.2 Beta (23C5047e) and it broke my installation of macFUSE.

**Comment by bfleischer** [comment]
@mnrv-ae I cannot confirm the the issue you are running into. Version 4.5.0 works fine for me on macOS 14.2 beta 3 (23C5047e). From which version of macOS did you upgrade to 14.2 beta 3?

**Comment by mnrv-ae** [comment]
> @mnrv-ae I cannot confirm the the issue you are running into. Version 4.5.0 works fine for me on macOS 14.2 beta 3 (23C5047e). From which version of macOS did you upgrade to 14.2 beta 3?

@bfleischer upgraded from 14,0 (after running /`usr/sbin/system_profiler SPInstallHistoryDataType`)

I've also noticed macFUSE appears on this list of installations 5 times, is this normal? 

![image](https://github.com/osxfuse/osxfuse/assets/74367063/68165705-1cfb-45c6-80d9-4f6ae1c61fff)

**Comment by Nathan187** [comment]
@mnrv-ae glad you were able to get it to work.  there is definitely a problem w/ veracrypt, macfuse and macos sonoma.  but they won't admit it for some reason.  annoying

**Comment by bfleischer** [comment]
@mnrv-ae I believe `SPInstallHistoryDataType` just displays the install history. macFUSE appearing multiple times is not out of the ordinary, if you ran the installer multiple times or updated macFUSE in the past. 

> Update: I did a re-install with homebrew this time instead of the .dmg binary, now I'm able to mount using rclone again, thankfully 

Homebrew downloads the .dmg file and runs the same installer. It should not make a difference whether you use Homebrew or download the .dmg and run the installer yourself.

**Comment by bfleischer** [comment]
@Nathan187 It's not about admitting whether there is an issue or not. Every complex software contains bugs. It's about being able to reproduce those issues/bug. Without being able to reproduce the issue you are running into, there is not much the VeraCrypt team or I can do to help you. Try providing as much information as possible when raising the issue with the VeraCrypt team.

**Comment by Nathan187** [comment]
@bfleischer thanks for the response.  i'm not the only one having problems and it's hard for me to believe that the teams mentioned haven't heard about it.  it's pretty straight forward i mean.  create a veracrypt contained (hidden or not) and then try to place a large file it in.  the operation will timeout/quite and say the location has disappeared.  i've seen reddit postings on this and other sources saying the same problem.  that's why i said out of frustration, no one wants to admit a problem

![Screenshot 2023-11-19 at 2 10 26 PM](https://github.com/osxfuse/osxfuse/assets/1022238/29d4b07a-8090-40a8-983f-b1f299f2659b)

**Comment by bfleischer** [comment]
@Nathan187 In theory, the issue you are describing could be caused by VeraCrypt, macFUSE or a bug in the macOS kernel. As far as I know VeraCrypt uses a complex setup to mount a container as a virtual volume. This could make debugging an issue like this rather complicated. There are many third-party file systems that are based on macFUSE, but I'm not aware of any other file systems exhibiting an issue like this. Using the LoopbackFS file system (the macFUSE reference file system), I have never run into an issue like this when copying large files (> 10 GB) to the volume.  This means the VeryCrypt team needs to investigate the issue. It might be caused by a deadlock in their file system code or the volume might have been unmounted due to a timeout, ... Assuming the VeryCrypt team is able to reproduce the issue, it should be easy for them to rule out deadlocks and timeouts and other possible causes in their code. If they come to the conclusion, that the issue is not caused by their file system code, they are free to open an issue here and provide the results of their investigation and ideally a minimalistic repro case.

**Comment by Nathan187** [comment]
@bfleischer thank you for your quick and concise response.  as with the op, i didn't start having problems until i started using sonoma.  i have to use an older mac i own if i'm moving files to volume, etc.  (macos catalina)   hopefully this will get fixed somehow.  just annoying but glad i have other macs i can use

**Comment by enoola** [comment]
Hi, @Nathan187 , same here. I am seeking for solutions since I updated to sonoma(~2 weeks)
- https://www.reddit.com/r/MacOS/comments/17wdo62/comment/ka9ujyo/?utm_source=reddit&utm_medium=web2x&context=3
As I mentionned in the last post (@phpeteur), there are 2 revivals of SSHFS project which I found so far 
https://github.com/tormodwill/macSSHFS .  But I am stuck with the compilation part mainly
Glad to read I am not alone @Nathan187   :P

**Comment by candasm** [comment]
I am running on Sonoma(14.0), m1 pro and MacFuse(4.5.0). I am able to read NTFS drives but I can't write anything on it. Honestly I didn't use NTFS drive for a long time but when I tried to backup some files I couldn't write to the external disk.

Can this be about enabling system extensions(System Settings> (Privacy & Security) > Enable System Extensions)?

**Comment by bfleischer** [comment]
@candasm You need an actual NTFS file system implementation that supports read/write access. One option ist using NTFS-3G which is built on FUSE. Getting things set up has become rather complicated on recent versions of macOS.

**Comment by candasm** [comment]
> @candasm You need an actual NTFS file system implementation that supports read/write access. One option ist using NTFS-3G which is built on FUSE. Getting things set up has become rather complicated on recent versions of macOS.

I installed with brew and I check the installed packages from package manager I had ntfs-3g package already. Also I tried manual installation from website but still not working. I can't write on my external disk.

**Comment by Nathan187** [comment]
@enoola I just updated my mac to sonoma 14.1.2 

so far, i haven't had any problems write to my volumes anymore.  sometimes it's slow but successful 

not sure if the update had anything to do with it.  let me know

**Comment by ultra-andy** [comment]
Hi @bfleischer, I've been having persistent problems using Veracrypt on MacOS Sonoma (currently 14.2 23C64). When I mount an encrypted container, open a file within the container (any text file), modify it and then try to save it, either the save operation fails immediately and the file appears corrupted, or it breaks a short time later. Access to the same container from a machine running Windows 10 works fine (it is stored in OneDrive), provided it has not yet been mounted in MacOS. Once it gets mounted in MacOS, all bets are off.

I have even tried restoring the OS (no luck) and even done a complete reset of the computer (an M1 Macbook Pro) and still get the same problem - so full reset, reinstall of MacFUSE 4.5.0, install of Veracrypt 1.26.7, still have a problem. I know there have been notes in recent Sonoma releases talking about (if I recall correctly) fixing issues with mounting encrypted containers in external SSD drives, which doesn't match my problem, but is a related topic.

It may not be related, but I have seen this message when opening the text file in the encrypted container: "The document 'filename.txt' is on a volume that does not support permanent version storage. You will not be able to access older versions of this document once you close it."

I still get problems even after copying the encrypted container to the MacOS desktop before mounting it, although in this case a finder window looking at the contents of the encrypted container spontaneously closed when I tried to save the file, and I got a message about "The document 'filename.txt' could not be saved.".

I know this is all a bit vague, so is there anything more specific I can do to obtain useful diagnostic information to pass on?
